src/main/java/org/assertj/core/api/AbstractCallableAssert.java

```diff
@@ -0,0 +1,49 @@
+package org.assertj.core.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.error.ShouldNotHaveThrown.shouldNotHaveThrown;
+
+import java.util.concurrent.Callable;
+
+import org.assertj.core.internal.Failures;
+
+public abstract class AbstractCallableAssert<SELF extends AbstractCallableAssert<SELF, V>, V>
+    extends AbstractAssert<SELF, Callable<V>> {
```

Comment:
@joel-costigliola I'm having second thoughts on extending `AbstractAssert` or `Assert` in general. I don't see any benefit on having the inherited assertions available, it will just pollute the code completion.

What if we would have a `ThrowingAssert` interface with two methods:
* `doesNotThrowAnyException`
* `hasThrownException`

My wish would be to use it here and also for `Runnable` related assertions (i.e., the existing `assertThatCode(ThrowingCallable)`), but I think the latter would be feasible only in version 4.

BTW in version 4 we could think about renaming `ThrowingCallable` to `ThrowingRunnable`.

Reply:
I think the only drawback would be not being able to describe the assertions but I'm happy we try that, we can make `ThrowingAssert` to implement `Descriptable`,  the value of a clean API overweight the need to reimplement describing assertions